src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -417,6 +438,9 @@ public int[] getDefaultTokens() {
             TokenTypes.BNOT,
             TokenTypes.POST_INC,
             TokenTypes.POST_DEC,
+            TokenTypes.BXOR,
+            TokenTypes.BOR,
+            TokenTypes.BAND,
```

Comment:
I am not sure that this is good default behavior.
Technically it is extra parentheses BUT compiler does not case and will optimize it as required, for human extra symbols do a lot of good job to show priority of execution. Nobody remember all this priorities, no need for human to even think of this, extra parentheses make it a way easier to read by all engineers.

extra parentheses are cheaper than mistakes caused by their absence.
At least we do not make behavior over strict by default.

@rnveach , @nrmancuso  , please vote.

Reply:
@romani I also think default behavior is not good

Reply:
I would prefer this check to report all unnecessary parentheses and have behavior similar to the same IDEA inspection, and allow user to ignore/suppress/remove tokens as they see fit, but I can understand not making these default.

Reply:
Ok, I we think it's not good default behavior, let's remove token from default collection. User can extend if he wants to be strict.
We can make example how to make it, to let it quickly be found.